system/lib/wasmfs/js_api.cpp

```diff
@@ -22,6 +22,8 @@ __wasi_fd_t wasmfs_create_file(char* pathname, mode_t mode, backend_t backend);
 // The buffer will also contain the file length.
 // Caller must free the returned pointer.
 void* _wasmfs_read_file(char* path) {
+  static_assert(sizeof(off_t) == 8, "File offset type must be 64-bit");
```

Comment:
Do we need this assert here.. I'm pretty sure if this ever changed all kind of stuff would break.

Reply:
Possibly the assert is a little redundant, but it's static so no overhead, and also it is helpful for the reader, I think, as it indicates this is a contract with the JS side that must not change.